### PR TITLE
feat(ivy): now supports SVG and MathML elements

### PR DESCRIPTION
**Pull request by benlesh**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently ivy does not support SVG elements

Issue Number: N/A

## What is the new behavior?

This adds support for SVG and MathML elements

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

This is breaking off just the SVG and MathML _element_ changes from #23899

### CODE REVIEW

File: packages/compiler/src/render3/view/template.ts [hunk 1]
```diff
@@ -314,11 +337,23 @@ export class TemplateDefinitionBuilder implements t.Visitor<void>, LocalResolver
...
+      this.addNamespaceInstruction(currentNamespace, element);
+    }
+
     this.instruction(
         this._creationCode, element.sourceSpan, R3.createElement, ...trimTrailingNulls(parameters));
 
     const implicit = o.variable(CONTEXT_NAME);
 
+
+
```
Comment (vicb) [inline packages/compiler/src/render3/view/template.ts]: nit: revert

File: packages/compiler/src/render3/r3_identifiers.ts [hunk 2]
```diff
@@ -17,6 +17,10 @@ export class Identifiers {
   static PATCH_DEPS = 'patchedDeps';
 
   /* Instructions */
+  static namespaceHTML: o.ExternalReference = {name: 'ɵNH', moduleName: CORE};
```
Comment (vicb) [inline packages/compiler/src/render3/r3_identifiers.ts]: Could you please update [environment.ts](https://github.com/angular/angular/blob/master/packages/core/src/render3/jit/environment.ts) with that.

File: packages/compiler/src/render3/view/template.ts (hunk 1 above)
Comment (benlesh) [reply packages/compiler/src/render3/view/template.ts]: done

File: packages/compiler/src/render3/r3_identifiers.ts (hunk 2 above)
Comment (benlesh) [reply packages/compiler/src/render3/r3_identifiers.ts]: done

**Review by vicb — APPROVED** [review]
Approved with the following modifs:
- update STATUS.md,
- squash the commits into a single one.

**Review by vicb — APPROVED** [review]

**Comment by benlesh** [comment]
Pull approve isn't green because I had to update the golden file.

File: packages/core/src/render3/instructions.ts [hunk 3]
```diff
@@ -575,7 +595,19 @@ export function elementStart(
       assertEqual(currentView.bindingIndex, -1, 'elements should be created before any bindings');
 
   ngDevMode && ngDevMode.rendererCreateElement++;
-  const native: RElement = renderer.createElement(name);
+
+  let native: RElement;
+
+  if (isProceduralRenderer(renderer)) {
+    native = renderer.createElement(name, _currentNamespace);
```
Comment (trotyl) [inline packages/core/src/render3/instructions.ts]: @benlesh @vicb This is breaking, the procedural renderer conceptually accepts the **namespace prefix**, not the **namespace URI**:

https://github.com/angular/angular/blob/668bfcec9d664e0d4c6caf35a52015fefbad4ab9/packages/platform-browser/src/dom/dom_renderer.ts#L112-L115

**Pull request closed without merging**